src/webui/api/torrentscontroller.cpp

```diff
@@ -737,8 +737,8 @@ void TorrentsController::setLocationAction()
     const QStringList hashes {params()["hashes"].split("|")};
     const QString newLocation {params()["location"].trimmed()};
 
-    // check if the location exists
-    if (newLocation.isEmpty() || !QDir(newLocation).exists())
+    // try to create the location if it does not exist
+    if (newLocation.isEmpty() || !QDir(newLocation).mkpath("."))
```

Comment:
nitpick: `mkpath('.')`

Reply:
'.' is a char but mkpath expects a string, so I am afraid that will not work.

Reply:
Shouldn't we raise some errors here? I.e. BadParams in case of `newLocation.isEmpty()` and something else if we couldn't create directory.
Or we can move to the default save folder in case of `newLocation.isEmpty()`...

Reply:
I added exception raising and tried to implement a mechanism for their displaying.